build.py

```
@@ -113,6 +116,43 @@ def _make_tmp_paths():
+    artifact = dependency['objects'][0]
+    output_path = source_tree / 'buildtools' / name / artifact['output_file']
+    if output_path.exists():
+        return
+
+    output_path.parent.mkdir(parents=True, exist_ok=True)
+    temporary_path = output_path.with_suffix(output_path.suffix + '.tmp')
+    url = 'https://storage.googleapis.com/{}/{}'.format(
+        dependency['bucket'], artifact['object_name']
+    )
+    try:
+        urllib.request.urlretrieve(url, temporary_path)
```

Ruff (0.16.2)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)